### PR TITLE
CTCP-1899-1: Add pagination to the drafts dashboard

### PR DESCRIPTION
**Pull request by Reece-Carruthers**

### CODE REVIEW

File: app/viewModels/paginationP5/Items.scala [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+case class Items(items: Seq[Item], firstItemDotted: Boolean, lastItemDotted: Boolean)
+
+object Items {
+
+  def apply(metaData: MetaData, href: String, additionalParams: Seq[(String, String)]): Items =
+    metaData match {
+      case MetaData(_, _, _, currentPage, totalNumberOfPages) =>
+        val itemList: Seq[Item] =
+          if (totalNumberOfPages < 6) {
```
Comment (ajchilts) [inline app/viewModels/paginationP5/Items.scala:27]: Is it perhaps worth putting the integer limits in this function into constants or maybe even defining them in app config so we can modify them?

File: app/views/components/PaginationP5.scala.html [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    @Html(viewModel.paginatedSearchResult())
+    </p>
+
+    <nav class="govuk-pagination" role="navigation" aria-label="results">
+
+        @viewModel.previous.map { p =>
+            <div class="govuk-pagination__prev">
+                <a id="previous" class="govuk-link govuk-pagination__link" href="@p.href" rel="prev">
+                    <svg class="govuk-pagination__icon govuk-pagination__icon--prev" xmlns="http://www.w3.org/2000/svg" height="13" width="15" aria-hidden="true" focusable="false" viewBox="0 0 15 13">
+                        <path d="m6.5938-0.0078125-6.7266 6.7266 6.7441 6.4062 1.377-1.449-4.1856-3.9768h12.896v-2h-12.984l4.2931-4.293-1.414-1.414z"></path>
```
Comment (ajchilts) [inline app/views/components/PaginationP5.scala.html:37]: I assume these are vector graphics for the view? Within the path 'd'.

File: conf/messages.en [hunk 3]
```diff
@@ -56,6 +56,9 @@ unauthorised.if.you.are.waiting.for.an.eori.number = You must have an account wi
 numberOfMovements.singular = Showing {0} result
 numberOfMovements.plural = Showing {0} results
 
+numberOfMovements.singular.withLRN = Showing {0} result matching {1}
```
Comment (ajchilts) [inline conf/messages.en:59]: Could this be?

`Showing 1 result matching {1}`

**Review by ajchilts — APPROVED** [review]
Fantastic piece of work. Nice 👍

File: app/models/DraftAvailability.scala [hunk 4]
```diff
@@ -22,8 +22,8 @@ object DraftAvailability {
 
   def apply(departureSummaries: Option[DeparturesSummary]): DraftAvailability = departureSummaries match {
     case Some(value) if value.userAnswers.nonEmpty => NonEmpty
-    case Some(_)                                   => Empty
-    case None                                      => Unavailable
+    case Some(value) if value.userAnswers.isEmpty  => Empty
```
Comment (jackheslop96) [inline app/models/DraftAvailability.scala:25]: Very minor but this case can only be hit if `value.userAnswers.isEmpty` is true so technically `if value.userAnswers.isEmpty` isn't needed

Comment (kenneth-rayner) [reply app/models/DraftAvailability.scala]: @jackheslop96  done... please re -review

**Review by ajchilts — APPROVED** [review]

**Review by jackheslop96 — APPROVED** [review]

**Pull request merged**